File: src/domain/monitor.ts
import type { AppSettings } from './settings'

/** Structured local-monitor / traffic status for Overview empty states. */
export type MonitorStatus =
  | 'disabled'
  | 'process_stopped'
  | 'port_conflict'
  | 'auth_failed'
  | 'timeout'
  | 'ok'
  | 'not_configured'

export interface FrpcTrafficResult {
  status: MonitorStatus
  body?: string
}

export interface ApplyLocalMonitorRequest {
  enabled: boolean
  addr: string
  port: number
  user?: string | null
  password?: string | null
}

export interface ApplyLocalMonitorResult {
  settings: AppSettings
  configPatched: boolean
  pendingRestart: boolean
}

File: src-tauri/src/domain/monitor.rs
use serde::{Deserialize, Serialize};

use super::settings::AppSettings;

/// Structured local-monitor / traffic status for Overview empty states.
#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum MonitorStatus {
    Disabled,
    ProcessStopped,
    PortConflict,
    AuthFailed,
    Timeout,
    Ok,
    NotConfigured,
}

#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct FrpcTrafficResult {
    pub status: MonitorStatus,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub body: Option<String>,
}

impl FrpcTrafficResult {
    pub fn status_only(status: MonitorStatus) -> Self {
        Self {
            status,
            body: None,
        }
    }

    pub fn ok(body: String) -> Self {
        Self {
            status: MonitorStatus::Ok,
            body: Some(body),
        }
    }
}

#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ApplyLocalMonitorRequest {
    pub enabled: bool,
    pub addr: String,
    pub port: u16,
    #[serde(default)]
    pub user: Option<String>,
    #[serde(default)]
    pub password: Option<String>,
}

#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
#[serde(rename_all = "camelCase")]
pub struct ApplyLocalMonitorResult {
    pub settings: AppSettings,
    pub config_patched: bool,
    pub pending_restart: bool,
}
